### ISSUE TITLE
Icons not loading correctly in emulsify styleguide

### ISSUE THREAD

**Issue by shaal** [fourkitchens/emulsify#255]
In the public styleguide of emulsify -
Atoms -> Images -> Icons
https://fourkitchens.github.io/emulsify/pattern-lab/public/?p=atoms-icons

Bars are not there, and the page loads with some errors -
```GET https://fourkitchens.github.io/dist/img/sprite/sprite.svg 404 ()```
and
```main-menu.js:25 Uncaught TypeError: Cannot read property 'getElementsByClassName' of null
    at main-menu.js:25
    at main-menu.js:18
(anonymous) @ main-menu.js:25
(anonymous) @ main-menu.js:18```

**Comment by ModulesUnraveled** [comment]
Hmmm... yeah. That sprite gets created on `npm start` (or `gulp icons`) and we're not running that (or committing the resulting `dist` directory) to Emulsify, so that file doesn't exist.

Not sure what the best way to solve this is.

@evanmwillhite any ideas?

**Comment by shaal** [comment]
But that same issue also happens on my local dev.
I downloaded latest Emulsify,
run 'yarn', 'gulp icons', 'yarn start',
and when I open the styleguide - same errors appear.

**Comment by ModulesUnraveled** [comment]
Hmmm... actually, it should work everywhere except for that pattern lab instance. did you try running npm install? There may be new requirements that you need to get.

"It works on my local" (tm)

**Comment by shaal** [comment]
Thank you!
`npm install` did solve it.
For some reason, I thought `npm install` is the same as running `yarn`

**Comment by ModulesUnraveled** [comment]
`yarn` uses the `yarn.lock` file by default, so it's pinned to specific versions  of stuff. Where as `npm install` just installs the latest.

If you could do me a favor, I'd like to debug this and see if the PR I just created will fix it for you and future yarn lovers:

- Delete your `node_modules` directory
- Run `yarn`

It should be broken

- Checkout this branch: https://github.com/fourkitchens/emulsify/pull/256
- Run `yarn`

It should be fixed

Let me know if that works.

**Comment by ModulesUnraveled** [comment]
Actually... I think I'm wrong. I couldn't get it to fail before updating the yarn.lock file... So I'm not sure what your issue was...

I'm going to close this issue since you're up and running now. Feel free to re-open if you have any other issues

**Comment by shaal** [comment]
(deleting node_modules was not enough to reset)

after `npm install`, icon was working like it should.
I deleted `node_modules` directory.
running `yarn` and 'yarn start` - icon was still working like it should.
so I deleted `emulsify` completely and downloaded it again.
now `yarn` and `yarn start` - icon is NOT working.

I deleted `emulsify` directory again.
checkout branch #256 (with the yarn.lock update)
running `yarn` and `yarn start`,  icon is NOT working.
running `yarn upgrade` and `yarn start`, icon is NOT working.
running `npm install` and `yarn start`, icon is NOT working.

I deleted `emulsify` directory again.
checkout latest emulsify dev branch
running `npm install` and `yarn start`, icon is NOT working.

now I don't understand how it worked the first time I ran `npm install`

**Comment by ModulesUnraveled** [comment]
Dang!

On a new install, try
- Clone emulsify and move into it
- `composer install`
- `npm install`
- `npm start`

**Comment by shaal** [comment]
did not work.

I even tried 
Install with NPM: `composer create-project fourkitchens/emulsify --stability dev --no-interaction emulsify && cd emulsify && npm install`
and it didn't work

**Comment by shaal** [comment]
I found a solution!

`cd web/themes/contrib/emulsify`
`php emulsify.php icontest`
`cd ../../custom/icontest`

edited package.json file, and replace 
```"emulsify-gulp": "^2.0.0",```
with
```"emulsify-gulp": "fourkitchens/emulsify-gulp#v2.1.0",```

`yarn`, `yarn start` - and icon IS working!

**Comment by ModulesUnraveled** [comment]
Dang. We just released emulsify-gulp 2.2.0 today, so `^2.0.0` should have been installing that. Can you try setting it to `"emulsify-gulp": "^2.2.0"` and see if that works?

**Comment by shaal** [comment]
yes, already checked that
with ^2.2.0 - icon is NOT working

**Comment by ModulesUnraveled** [comment]
Weird, that worked for me this morning.

can you try `"emulsify-gulp": "fourkitchens/emulsify-gulp#v2.2.0",`

**Comment by shaal** [comment]
that is what I tried
"emulsify-gulp": "fourkitchens/emulsify-gulp#v2.1.0",

**Comment by ModulesUnraveled** [comment]
Yeah, I'm wondering if the caret thing isn't getting the latest for some reason, and is installing an old version instead.

I'd like you to try pinning to v2.2.0 to see if that works.

**Comment by shaal** [comment]
can you please give me the actual code to put there, so I do it the exact way you want it to be tested?

**Comment by ModulesUnraveled** [comment]
Ah, before you edited your comment, it said `...#v2.1.0`, I wanted you to try `...#v2.2.0`. It looks like that didn't work for you either.

We're going to do some more testing internally.

I'm closing this for now since all of our tests originally worked, and it could be anything from your npm cache, or something about your local environment.

We'll stay on top of it, and see if anyone else reports an issue in the meantime.

**Comment by grshane** [comment]
I tested this and am seeing the same thing as @shaal. 
emulsify-gulp: 2.2.0

Node: 8.11.1

**Comment by ModulesUnraveled** [comment]
Weird. This is working fine for all of us internally...

Reopening so that we can track what's going on

**Comment by ModulesUnraveled** [comment]
@grshane Are you on Linux by chance?

**Comment by ModulesUnraveled** [comment]
@shaal would you mind reviewing this PR? I think our best bet is to go ahead and pin that version for now: https://github.com/fourkitchens/emulsify/pull/259

**Comment by shaal** [comment]
I just finished checking #259 
Unfortunately, it still doesn't solve the problem.
Steps:
After I cloned that branch,
`composer install`
`yarn install`
`yarn start`
Icons are not working.

BUT, when I run `yarn upgrade`, icons DO work.
so I compared the directories of icons-fix branch and icons-fix after running yarn upgrade.
I couldn't find any files that are different (except node_modules, etc)

if this is helpful -
in this directory - /node_modules/emulsify-gulp 
these are the 3 files that are different
gulp-config.js
index.js
package.json
